catalogs/catalog-fileset/src/main/java/com/datastrato/gravitino/catalog/file/HadoopCatalogOperations.java

```diff
@@ -108,23 +153,168 @@ public NameIdentifier[] listSchemas(Namespace namespace) throws NoSuchCatalogExc
+                    .withCreateTime(Instant.now())
+                    .build())
+            .build();
+    try {
+      store.put(schemaEntity, true /* overwrite */);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to create schema " + ident.name(), ioe);
+    }
+
+    return new HadoopSchema.Builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withProperties(schemaEntity.properties())
+        .withAuditInfo(schemaEntity.auditInfo())
+        .build();
   }
 
   @Override
   public Schema loadSchema(NameIdentifier ident) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(schemaEntity.comment())
+          .withProperties(schemaEntity.properties())
+          .withAuditInfo(schemaEntity.auditInfo())
+          .build();
+
+    } catch (NoSuchEntityException exception) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", exception);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to load schema " + ident.name(), ioe);
+    }
   }
 
   @Override
   public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
       throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+    try {
+      if (!store.exists(ident, Entity.EntityType.SCHEMA)) {
+        throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist");
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to check if schema " + ident.name() + " exists", ioe);
+    }
+
+    try {
+      SchemaEntity entity =
+          store.update(
+              ident,
+              SchemaEntity.class,
+              Entity.EntityType.SCHEMA,
+              schemaEntity -> updateSchemaEntity(ident, schemaEntity, changes));
+
+      return new HadoopSchema.Builder()
+          .withName(ident.name())
+          .withComment(entity.comment())
+          .withProperties(entity.properties())
+          .withAuditInfo(entity.auditInfo())
+          .build();
+
+    } catch (IOException ioe) {
+      throw new RuntimeException("Failed to update schema " + ident.name(), ioe);
+    } catch (NoSuchEntityException nsee) {
+      throw new NoSuchSchemaException("Schema " + ident.name() + " does not exist", nsee);
+    } catch (AlreadyExistsException aee) {
+      throw new RuntimeException("Schema " + ident.name() + " already exists", aee);
+    }
   }
 
   @Override
   public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
-    throw new UnsupportedOperationException();
+    try {
+      SchemaEntity schemaEntity = store.get(ident, Entity.EntityType.SCHEMA, SchemaEntity.class);
```

Comment:
do we need to delete the schema from the store?

Reply:
Yes, we will, it is implemented in `CatalogOperationDispatcher#dropSchema`.